wurst/math/MatricesTests.wurst

```diff
@@ -0,0 +1,297 @@
+    test.transpose().assertEquals(expected)
+
+@Test function test22Determinant()
+    mat2(1, 2, 3, 4).determinant().assertEquals(-2, 0.002)
+    mat2(15, 5, 6, 2).determinant().assertEquals(0, 0.002)
+    mat2(-2, 128, 37, -42).determinant().assertEquals(-4652, 0.002)
+
+@Test function test22Inverse()
+    var test = mat2(1, 2, 3, 4)
+    var inverse = test.inverse()
+    inverse.success.assertTrue()
+    (test * inverse.matrix).assertEquals(IDENTITY22)
+    test = mat2(15, 5, 6, 2)
+    inverse = test.inverse()
+    inverse.success.assertFalse()
+    (test * inverse.matrix).assertEquals(ZERO22)
+    test = IDENTITY22
+    inverse = test.inverse()
+    inverse.success.assertTrue()
+    test.assertEquals(inverse.matrix)
+
+@Test function test22Angle2Mat()
+    90 .asAngleDegrees().toMatrix().assertEquals(mat2(0, -1, 1, 0))
+    (-90).asAngleDegrees().toMatrix().assertEquals(mat2(0, 1, -1, 0))
+    (225).asAngleDegrees().toMatrix().assertEquals(mat2(-0.707, 0.707, -0.707, -0.707))
+
+@Test function test22Scaling()
+    vec2(0.5, 0.5).toScaling().assertEquals(mat2(0.5, 0, 0, 0.5))
+    vec2(1.5, 7.5).toScaling().assertEquals(mat2(1.5, 0, 0, 7.5))
+    (vec2(2, 2).toScaling() * vec2(5, 16)).assertEquals(vec2(10, 32))
+    
+/* =========== 3X3 MATRIX =========== */
+
+function mat3.assertEquals(mat3 expected)
+    let delta = 0.002
+    let compare = mat3( (this.m00 - expected.m00).abs(), (this.m01 - expected.m01).abs(), (this.m02 - expected.m02).abs(),
+                        (this.m10 - expected.m10).abs(), (this.m11 - expected.m11).abs(), (this.m12 - expected.m12).abs(),
+                        (this.m20 - expected.m20).abs(), (this.m21 - expected.m21).abs(), (this.m22 - expected.m22).abs())
+    let msg = "Expected {0} <{1}>, Actual <{2} with delta {3}>"
+    if compare.m00 > delta
+        testFail(msg.format("m00", expected.m00.toString(), this.m00.toString(), delta.toString()))
+    if compare.m01 > delta
+        testFail(msg.format("m01", expected.m01.toString(), this.m01.toString(), delta.toString()))
+    if compare.m02 > delta
+        testFail(msg.format("m02", expected.m02.toString(), this.m02.toString(), delta.toString()))
+
+    if compare.m10 > delta
+        testFail(msg.format("m10", expected.m10.toString(), this.m10.toString(), delta.toString()))
+    if compare.m11 > delta
+        testFail(msg.format("m11", expected.m11.toString(), this.m11.toString(), delta.toString()))
+    if compare.m12 > delta
+        testFail(msg.format("m12", expected.m12.toString(), this.m12.toString(), delta.toString()))
+
+    if compare.m20 > delta
+        testFail(msg.format("m20", expected.m20.toString(), this.m20.toString(), delta.toString()))
+    if compare.m21 > delta
+        testFail(msg.format("m21", expected.m21.toString(), this.m21.toString(), delta.toString()))
+    if compare.m22 > delta
+        testFail(msg.format("m22", expected.m22.toString(), this.m22.toString(), delta.toString()))
+
+function vec3.assertEquals(vec3 expected)

```

Comment:
should go into the vector package I assume

Reply:
Hmm... Would it better to move mat3.assertEquals from MatricesTests into Matrices as well? And do the same for each similar function in the future.